### ISSUE TITLE
Bug in debug routine `BPLUSTREE_TYPE::ToGraph`

### ISSUE THREAD

**Issue by l-iberty** [cmu-db/bustub#117]
`BPLUSTREE_TYPE::ToGraph` defined here [https://github.com/cmu-db/bustub/blob/master/src/storage/index/b_plus_tree.cpp#L282](https://github.com/cmu-db/bustub/blob/master/src/storage/index/b_plus_tree.cpp#L282)

```cpp
// Print leaves
for (int i = 0; i < inner->GetSize(); i++) {
  auto child_page = reinterpret_cast<BPlusTreePage *>(bpm->FetchPage(inner->ValueAt(i))->GetData());
  ToGraph(child_page, bpm, out);
  if (i > 0) {
    auto sibling_page = reinterpret_cast<BPlusTreePage *>(bpm->FetchPage(inner->ValueAt(i - 1))->GetData());
    if (!sibling_page->IsLeafPage() && !child_page->IsLeafPage()) {
      out << "{rank=same " << internal_prefix << sibling_page->GetPageId() << " " << internal_prefix
          << child_page->GetPageId() << "};\n";
    }
  }
}
```

We can see from this segment of codes that `sibling_page` is not UNPINED after `FetchPage` call. This may cause some bugs.

**Comment by apavlo** [comment]
@l-iberty Can you send a PR with the fix?

**Comment by lmwnshn** [comment]
Closed by #118.
